x/mm2/keeper/genesis.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/skip-mev/slinky/x/mm2/types"
+)
+
+// InitGenesis initializes the genesis state. Panics if there is an error.
+// Any modules that integrate with x/marketmap must set their InitGenesis to occur before the x/marketmap
+// module's InitGenesis.  This is so that logic any consuming modules may want to implement in AfterMarketGenesis
+// will be run properly.
+func (k *Keeper) InitGenesis(ctx sdk.Context, gs types.GenesisState) {
+	// validate the genesis
+	if err := gs.ValidateBasic(); err != nil {
+		panic(err)
+	}
+
+	for _, market := range gs.MarketMap.Markets {
+		if err := k.createMarket(ctx, market); err != nil {
+			panic(err)
+		}
+	}
```

Comment:
Why not `CreateMarkets`? That does the extra state validation we probably want no?

Reply:
the genesis will already be validated because we have all of the information needed.   We also want to set `LastUpdated` to the correct value set in the genesis